apps/src/sites/studio/pages/blocks/edit.js

```diff
@@ -27,7 +27,17 @@ $(document).ready(() => {
   });
 
   initializeCodeMirrorForJson('block_config', { onChange });
-  initializeCodeMirror('block_helper_code', 'javascript');
+  const helperCodeEditor = initializeCodeMirror('block_helper_code', 'javascript');
+
+  document.querySelector('#block_submit').addEventListener('click', event => {
+    try {
+      const code = helperCodeEditor.getValue();
+      eval(`(function () { ${code} })`); // eslint-disable-line no-eval
```

Comment:
Given that codemirror is already doing syntax highlighting, is there any chance it can also expose the known syntax errors programmatically?